demos/python_demos/formula_recognition_demo/formula_recognition_demo.py

```diff
@@ -140,90 +208,87 @@ def build_argparser():
-                                                     args.tgt_layer: tgt
-                                                     }
-                                             )
-
-            dec_states_h = dec_res[args.dec_st_h_t_layer]
-            dec_states_c = dec_res[args.dec_st_c_t_layer]
-            output = dec_res[args.output_layer]
-            logit = dec_res[args.logit_layer]
-            logits.append(logit)
-            tgt = np.array([[np.argmax(logit, axis=1)]])
-
-            if tgt[0][0][0] == END_TOKEN:
-                break
-        if args.perf_counts:
-            log.info("Encoder performance statistics")
-            print_stats(exec_net_encoder)
-            log.info("Decoder performance statistics")
-            print_stats(exec_net_decoder)
-
-        logits = np.array(logits)
-        logits = logits.squeeze(axis=1)
-        targets = np.argmax(logits, axis=1)
-        if args.output_file:
-            with open(args.output_file, 'a') as output_file:
-                output_file.write(rec['img_name'] + '\t' +  vocab.construct_phrase(targets) + '\n')
+    args = build_argparser().parse_args()
+    assert bool(args.interactive) != bool(args.input), "Choose only one option from [--interactive] and [-i/--input]"
+    model = Model(args)
+    if not args.interactive:
+        non_interactive_demo(model, args)
+        return
+
+    *_, height, width = model.encoder.input_info['imgs'].input_data.shape
+    prev_text = ''
+    demo = InteractiveDemo((height, width), resolution=args.resolution)
+    capture = create_capture(args, demo.resolution)
+    while True:
+        ret, frame = capture.read()
+        bin_crop = demo.get_crop(frame)
+        model_input = prerocess_crop(bin_crop, (height, width), preprocess_type=args.preprocessing_type)
+        frame = demo.put_crop(frame, model_input)
+        model_res = model.infer_async(model_input)
+        if not model_res:
+            phrase = prev_text
         else:
-            print("Image name: {}\nFormula: {}\n".format(rec['img_name'], vocab.construct_phrase(targets)))
-
+            logits, targets = model_res
+            prob = calculate_probability(logits)
+            log.info("Confidence score is %s", prob)
+            if prob >= args.conf_thresh ** len(logits):
+                log.info("Prediction updated")
+                phrase = model.vocab.construct_phrase(targets)
+            else:
+                log.info("Confidence score is low, prediction is not complete")
+                phrase = ''
+        frame = demo.draw(frame, phrase)
+        prev_text = phrase
+        cv.imshow('Press q to quit.', frame)
+        key = cv.waitKey(1) & 0xFF
+        if key == ord('q'):
+            break
+        elif key == ord('o'):
+            demo.resize_window("decrease")
+        elif key == ord('p'):
+            demo.resize_window("increase")
+
+    capture.release()
+    cv.destroyAllWindows()
```

Comment:
```suggestion
```
They will be destroyed any way.

Reply:
Deleted this. I agree with you, but I followed a style guide that says release resources as far as we do not need them.